### PR TITLE
added logic contract deployment optimisation

### PR DESCRIPTION
**Pull request by shivamgarglab49**

### CODE REVIEW

File: e2e-test/step-definitions/MultiSigDAOTest.ts [hunk 1]
```diff
@@ -443,9 +443,6 @@ export class MultiSigDAOSteps {
 
   @then(/User verify contract logic address is updated/, undefined, 30000)
```
Comment (amandeeprehal-lab49) [inline e2e-test/step-definitions/MultiSigDAOTest.ts:444]: Do we need to do this?

File: deployment/checkAndDeploy.ts [hunk 2]
```diff
@@ -1,57 +1,59 @@
...
-import { ContractId } from "@hashgraph/sdk";
-import { clientsInfo } from "../utils/ClientManagement";
 import { ContractService } from "./service/ContractService";
-import { DeployedContract } from "./model/contract";
+import { main as initializeContracts } from "./scripts/initializeContracts";
 
-const web3 = new Web3();
-const deployment = new Deployment();
+const NON_PROXY_CONTRACTS_WITH_LOWER_CASE_NAME =
+  ContractMetadata.NON_PROXY_CONTRACTS.map((item: string) =>
```
Comment (amandeeprehal-lab49) [inline deployment/checkAndDeploy.ts:9]: You can move it to ContractMetadata class.  NON_PROXY_CONTRACTS_WITH_LOWER_CASE_NAME It is going to remain the same so you can create static constant if you like.

File: deployment/checkAndDeploy.ts [hunk 3]
```diff
@@ -1,57 +1,59 @@
...
+  const allContractsToDeploy =
+    await contractMetadata.getAllChangedContractNames(service);
+  if (allContractsToDeploy.length === 0) {
+    console.log(`No contract for auto upgrade available`);
+    return;
   }
-}
+  console.log(`Eligible contracts for auto upgrade:`, allContractsToDeploy);
+
+  const proxyContractsToDeploy = allContractsToDeploy.filter(
```
Comment (amandeeprehal-lab49) [inline deployment/checkAndDeploy.ts:26]: Pls move it to ContractMetadata `allContractsToDeploy.filter(`

File: deployment/checkAndDeploy.ts [hunk 4]
```diff
@@ -1,57 +1,59 @@
...
+    (item: string) => !NON_PROXY_CONTRACTS_WITH_LOWER_CASE_NAME.includes(item)
+  );
+
+  const nonProxyContractsToDeploy = allContractsToDeploy.filter(
```
Comment (amandeeprehal-lab49) [inline deployment/checkAndDeploy.ts:30]: Same here

File: e2e-test/step-definitions/MultiSigDAOTest.ts (hunk 1 above)
Comment (shivamgarglab49) [reply e2e-test/step-definitions/MultiSigDAOTest.ts:406]: yes

**Pull request closed without merging**